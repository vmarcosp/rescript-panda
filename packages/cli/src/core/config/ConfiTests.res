open Vitest

describe("StyleSystem", () => {
  test(".extractAndMerge(...)", _ => {
    open Expect

    let theme: Config.extensibleTheme = %raw(`{
      extend: {
        colors: {
          primary: { 300: { value: '#c3c3c3' } },
          secondary: { dark: { 100: { value: '#cecece'} }, light: { value: '#fefefe' } }
        }
      },
      colors: {
        primary: { 100: { value: '#fafafa' }, 200: { value: '#000000' } },
        secondary: { 100: { value: '#fc3666' }, dark: { 200: { value: '#d3d3d3' } } }
      }
    }`)

    let preset: Config.preset = %raw(`{
      name: 'test-preset-1',
      theme: {
        colors: {
          neutral: { 100: { value: '#fafafa' }, 200: { value: '#000000' } },
          warning: { 100: { value: '#fc3666' }, dark: { 200: { value: '#d3d3d3' } } }
        }
      }
    }`)

    expect(Config.extractAndMerge(theme, Some([preset])))->toMatchSnapshot
    expect(Config.extractAndMerge(theme, None))->toMatchSnapshot
  })
})
