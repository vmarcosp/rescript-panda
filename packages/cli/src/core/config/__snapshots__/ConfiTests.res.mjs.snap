// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StyleSystem > .extractAndMerge(...) 1`] = `
{
  "colors": {
    "neutral": {
      "100": {
        "value": "#fafafa",
      },
      "200": {
        "value": "#000000",
      },
    },
    "primary": {
      "100": {
        "value": "#fafafa",
      },
      "200": {
        "value": "#000000",
      },
      "300": {
        "value": "#c3c3c3",
      },
    },
    "secondary": {
      "100": {
        "value": "#fc3666",
      },
      "dark": {
        "100": {
          "value": "#cecece",
        },
        "200": {
          "value": "#d3d3d3",
        },
      },
      "light": {
        "value": "#fefefe",
      },
    },
    "warning": {
      "100": {
        "value": "#fc3666",
      },
      "dark": {
        "200": {
          "value": "#d3d3d3",
        },
      },
    },
  },
}
`;
